### DISCUSSION TITLE
Include vertical distance in `/route` output for level traversing routes

### DISCUSSION THREAD

**Discussion by chrstnbwnkl** [valhalla/valhalla#5390] [Q&A]
The `levels` information in EdgeInfo could help derive the total vertical distance traveled for a route (or maneuver), when the route includes level changes (via escalator, elevator or steps). Right now there is no notion of vertical distance except for elevation, which is not meant for these more indoor type vertical changes. 

There also does not seem to be a designated tag in OSM to hold this information on steps or elevators. What comes to my mind would be something like `height:level` or `level:height`, neither of which is used at all according to taginfo. 

I guess what I'm asking is whether this is something that's interesting for the public repo even though OSM has no way to properly tag it (yet).

**Comment by kevinkreiser** [comment]
i personally dont think it has a use in the router. you could make a stretch and say that you could vary the time spend in the elevator (when we add that to the transition cost) based on the height of the levels traversed but its going to be so negligable its hardly worth it. to me this information is way way more useful in visual modeling, ie vector tiles or whatever other cad type systems exist for such data. at the same time its a tiny amount of data (non in osm at the moment apparently) so it could be supported i just dont think there really is a compelling argument for it when its related to an elevator.

when we think about routing and costing we think about distance traveled and indeed vertical distance does count but when it comes to elevators it doesnt matter. now if we had a use case like ladders or steps or something it could make more sense but really ladders and steps dont really vary much on how much height a step or rung of a ladder contributes when you use them. in other words they are pretty standardized. in that way a simple multiplier when steps come into the route into the route is good enough. ladders typically dont vary much in height (and we dont su pport them) but if they did vary significantly and they were something that people did a lot it would make sense. frankly though its just not prevalent enough. similarly elevators dont vary that much in terms of time or effort so differentiating height is like splitting hairs for routing. to me, again, the desire for such a thing is more visual than routing.

just my two cents!
